File: medium/letter-case-permutation.js
/**
 * @param {string} s
 * @return {string[]}
 */
var letterCasePermutation = function(s) {

    const arr= []
    
    const backtrack = (current, index)=>{
        if (index == s.length){
            arr.push(current)
            return
        }

        backtrack(current + s[index].toLowerCase() , index+1)

       if (/[a-zA-Z]/.test(s[index])) {
            backtrack(current + s[index].toUpperCase(), index+1)
        }
    }

    backtrack('',0)


    return arr
};

File: medium/letter-case-permutation.go
package medium
import "unicode"



func letterCasePermutation(s string) []string {
    var arr []string


    var backtrack func(current string, index int)
    backtrack=func(current string , index int ){
       if index==len(s){
           arr=append(arr,current)
           return
       }

       backtrack(current + string(unicode.ToLower(rune(s[index]))), index+1)

       if unicode.IsLetter(rune(s[index])){
           backtrack(current + string(unicode.ToUpper(rune(s[index]))), index+1)
       }
    }

    backtrack("", 0)

    return arr
}

func main(){
    letterCasePermutation("abcd")
}







